core/tests/test_utils.py

```diff
@@ -564,6 +594,37 @@ def tearDown(self):
     def _get_all_queue_names(self):
         return [q['name'] for q in self.taskqueue_stub.GetQueues()]
 
+    @contextlib.contextmanager
+    def urlfetch_mock(
+            self, content='', status_code=200, headers=None):
+        """Enables the custom urlfetch mock (URLFetchServiceMock) within the
+        context of a 'with' statement.
+
+        This mock is currently used for signup to prevent external HTTP
+        requests to fetch the Gravatar profile picture for new users.
```

Comment:
Also, just a thought -- the e2e tests will be affected, too. I'm not sure there's much we can do about that, though...